modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/walkthrough/Walkthrough.js

```diff
@@ -0,0 +1,422 @@
+	 * the component will be updated.
+	 */
+	useEffect(() => {
+		setCurrentAlignment(defaultPositioning);
+	}, [defaultPositioning]);
+
+	const align = useCallback(() => {
+		if (popoverVisible && popoverRef.current) {
+			const points = ALIGNMENTS_MAP[currentAlignment];
+
+			const alignment = doAlign({
+				offset: [OVERLAY_OFFSET_X, OVERLAY_OFFSET_Y],
+				overflow: {
+					adjustX: true,
+					adjustY: true,
+				},
+				points,
+				sourceElement: popoverRef.current,
+				targetElement: trigger,
+			});
+
+			const alignmentString = alignment.points.join('');
+
+			const pointsString = points.join('');
+
+			if (alignment.overflow.adjustX) {
+				setCurrentAlignment(ALIGNMENTS_GUESS_MAP[alignmentString]);
+			}
+			else if (pointsString !== alignmentString) {
+				setCurrentAlignment(ALIGNMENTS_INVERSE_MAP[alignmentString]);
+			}
+
+			if (!darkbg && previousTrigger !== trigger) {
+				trigger.classList.add('lfr-walkthrough-element-shadow');
+
+				previousTrigger.classList.remove(
+					'lfr-walkthrough-element-shadow'
+				);
+			}
+		}
+	}, [
+		currentAlignment,
+		darkbg,
+		popoverRef,
+		popoverVisible,
+		previousTrigger,
+		trigger,
+	]);
+
+	useEffect(() => {
+		align();
+	}, [align]);
+
+	useObserveRect(align, popoverRef?.current);
+
+	return (
+		<>
+			{!popoverVisible && (
+				<Hotspot
+					onHotspotClick={() => setPopoverVisible(true)}
+					ref={hotspotRef}
+					trigger={trigger}
+				/>
+			)}
+
+			{darkbg && (
+				<WalkthroughOverlay
+					popoverVisible={popoverVisible}
+					trigger={trigger}
+				/>
+			)}
+
+			{popoverVisible && (
+				<ReactPortal>
+					<ClayPopover
+						alignPosition={currentAlignment}
+						closeOnClickOutside={closeOnClickOutside}
+						displayType="secondary"
+						header={
+							<ClayLayout.ContentRow
+								noGutters
+								verticalAlign="center"
+							>
+								<ClayLayout.ContentCol expand>
+									<span>{`Step ${
+										stepIndex + 1
+									} of ${stepLength}: ${title}`}</span>
+								</ClayLayout.ContentCol>
+
+								{closeable && (
+									<ClayLayout.ContentCol>
+										<ClayButtonWithIcon
+											aria-label={Liferay.Language.get(
+												'close'
+											)}
+											className="close"
+											displayType="unstyled"
+											onClick={() =>
+												setPopoverVisible(false)
+											}
+											small
+											symbol="times"
+										/>
+									</ClayLayout.ContentCol>
+								)}
+							</ClayLayout.ContentRow>
+						}
+						onShowChange={setPopoverVisible}
+						ref={popoverRef}
+						show={popoverVisible}
+						size="lg"
+					>
+						<div
+							dangerouslySetInnerHTML={{
+								__html: content,
+							}}
+						></div>
+
+						<ClayLayout.ContentRow noGutters verticalAlign="center">
+							{skippable && (
+								<ClayLayout.ContentCol expand>
+									<ClayCheckbox
+										label={Liferay.Language.get(
+											'do-not-show-me-this-again'
```

Comment:
This functionallity was not implemented because it should be persisted on back-end side. Leaving this reminder of monday's Diego version and maybe if @marcoscv-work or @dsanz knows if this thing was already mapped on requirements and would help Diego's monday version :)